### PR TITLE
Lidarr 

### PR DESCRIPTION
**Pull request by Safihre**
_Linked issues:_ #3135

### Checklist
- [x] Build rule `all-supported` completed [successfully](https://travis-ci.org/Safihre/spksrc/builds/395916143)
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

File: spk/lidarr/Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+SERVICE_PORT_TITLE = $(DISPLAY_NAME)
+
+# Admin link for in DSM UI
+ADMIN_PORT = $(SERVICE_PORT)
+
+WIZARDS_DIR = src/wizard/
+
+POST_STRIP_TARGET = lidarr_extra_install
+
+BUSYBOX_CONFIG = usrmng
```
Comment (ymartin59) [inline spk/lidarr/Makefile]: No need for busybox from my point of view

File: spk/lidarr/Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+SPK_NAME = lidarr
+SPK_VERS = $(shell date +%Y%m%d)
+SPK_REV = 1
+SPK_ICON = src/lidarr.png
+
+REQUIRED_DSM = 5.0
+UNSUPPORTED_ARCHS = $(PPC_ARCHES)
+
+DEPENDS = cross/curl cross/mediainfo cross/sqlite cross/lidarr
```
Comment (ymartin59) [inline spk/lidarr/Makefile]: mediainfo is a really large package... If lidarr invokes it as a fork process, I would recommend a package dependency and configuration of PATH to have commands available to the application
... if it was possible to do the same with curl and sqlite, that package may be non-arch or I am wrong?

File: spk/lidarr/src/service-setup.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
+PATH="${SYNOPKG_PKGDEST}/bin:${PATH}"
```
Comment (ymartin59) [inline spk/lidarr/src/service-setup.sh]: if ok with package dependency of mediainfo, add `/var/packages/mediainfo/target/bin`

File: spk/lidarr/src/service-setup.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+service_preupgrade ()
+{
+    # Is Installed Lidarr Binary Ver. >= SPK Lidarr Binary Ver.?
+    CUR_VER=$(${MONO_PATH}/monodis --assembly ${LIDARR} | grep "Version:" | awk '{print $2}')
+    echo "Installed Lidarr Binary: ${CUR_VER}" >> ${INST_LOG}
+    SPK_VER=$(${MONO_PATH}/monodis --assembly ${SPK_LIDARR} | grep "Version:" | awk '{print $2}')
+    echo "Requested Lidarr Binary: ${SPK_VER}" >> ${INST_LOG}
+    if [ "${CUR_VER//.}" -ge "${SPK_VER//.}" ]; then
+        echo 'KEEP_CUR="yes"' > ${CONFIG_DIR}/KEEP_VAR
```
Comment (ymartin59) [inline spk/lidarr/src/service-setup.sh]: I wonder if you may append your variables to generic support file: https://github.com/SynoCommunity/spksrc/blob/master/mk/spksrc.service.installer#L9
... and maybe this support needs improvements for "upgrade" ...

File: spk/lidarr/src/wizard/install_uifile [hunk 5]
```diff
@@ -0,0 +1,11 @@
+[{
+    "step_title": "Starting and updating Lidarr",
+    "items": [{
+        "desc": "The first time Lidarr is started it might take a few moments for the interface to become available!<br><br>Keep Lidarr up-to-date by using Lidarr's built-in updater.<br>Navigate to System>Updates in the Lidarr UI."
+    }]
+},{
+    "step_title": "Attention! DSM Permissions",
+    "items": [{
+        "desc": "Permissions for all download-related packages are managed with the group <b>'sc-download'</b> in DSM.<br>The group 'users' is no longer used as of DSM 6.<br>Package user will not appear on most UI settings."
```
Comment (ymartin59) [inline spk/lidarr/src/wizard/install_uifile]: Please report link to Permission-Management wiki page I have added to other packages (same in _fre and for upgrade wiazrds)

**Comment by Safihre** [comment]
I will try to publish myself if that's oke? :)

**Comment by ymartin59** [comment]
Packages have been uploading in repository. Why not activated yet? What is the status?

**Comment by ymartin59** [comment]
There are too much builds from my point of view: https://synocommunity.com/admin/buildview/?flt0_0=lidarr
According to https://github.com/SynoCommunity/spksrc/wiki/Update-Policy
you should limit to `make publish-arch-qoriq-5.2 publish-arch-x64-5.2 publish-arch-evansport-5.2 publish-arch-88f6281-5.0 publish-arch-alpine-5.2 \
     publish-arch-armada370-5.2 publish-arch-armada375-5.2 publish-arch-armadaxp-5.2 publish-arch-armada38x-5.2 \
     publish-arch-comcerto2k-5.2 publish-arch-monaco-5.2 publish-arch-ppc853x-5.2 publish-arch-rtd1296-6.1`

**Comment by Safihre** [comment]
The wiki pages says specially that packages that depend on `mediainfo` need to have seperate compiled ones for DSM 5 and 6: https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#compatibility-issue
It specifically lists Radarr and Sonarr, so I assumed it applies to Lidarr too. Does it not? 
I did not publish because for some reason it lists on that page half of the 6.1 packages as being 3.1, do you know why?

**Comment by ymartin59** [comment]
Correct... and strange firmware version, that is right. Probably a trouble with `TC_OS_MIN_VER`.
Not sure about it, but I guess it may come from that toolchain code part: #3371
Other proposal: in `local.mk`, set `SUPPORT_URL=https://github.com/SynoCommunity/spksrc/issues` (I should document it too...)

**Comment by Safihre** [comment]
We can edit the main `Makefile` for that, at the bottom.
Does the version parsing matter? I guess it does for what files it offers to the Package centers. Wanted to double check the generated info/package files first, but maybe you know if it's a problem in spkrepo? 
I also noticed that recent packages in the list (before Lidarr) are not listed as "Startable" even though they are. Like `sickbeard-custom`.

**Comment by ymartin59** [comment]
There are two problems with recent changes in INFO generation: "startable" has been removed because it defaults to "true"... but `spkrepo` has not been "improved" to consider it as defaults.
Then "os_min_ver" is not know yet by `spkrepo`... so even if PR fixes INFO content, I guess generated packages are no longer deliverable to DSM...

**Comment by ymartin59** [comment]
@Safihre I quickly patched `spkrepo` https://github.com/SynoCommunity/spkrepo/pull/20 and seems to be OK. I am do some publishing to confirm. You should delete previous lidarr builds.

**Comment by Safihre** [comment]
@ymartin59 Thanks. Removed builds and will now trigger one again with the new code.

**Comment by Safihre** [comment]
Re-added, does it look good now @ymartin59 ?

**Comment by ymartin59** [comment]
Correct. "startable" flag is still wrong because "version" has not be created. But does not matter. A good shortcut to remove all builds is to delete associate "version". OK so you can activate all builds for that "version" and check in Package Center (notice it may to time to refresh)

**Comment by ymartin59** [comment]
I leave for two weeks vacation.

**Pull request merged**